[FIX] 11.0 l10n_es_aeat_sii: amount total tax line incorrect

RodrigoBM opened the pull request:
Este FIX soluciona la información enviada al SII en las facturas que tienen una moneda diferente. Si tienes una factura con:
Base imponible: 605,16£
Cuota de IVA: 127,08£

Nuestra empresa al ser española, tenemos que informar de estos importes en euros:
Base imponible: 0,899030 / 605,16£ = 673,13€.
Cuota de IVA: 0,899030 / 127,08£ = 141,35€.

Sin este FIX estaríamos enviando al SII un importe total de (673,13€ + 127,08£) 800,21€. Esto no es correcto, ya que el importe de 673,13 se informa en euros pero estamos añadiendo el IVA en 127,08£.

Con esta corrección el importe que nos queda por añadir como tasa de IVA es la moneda de la empresa utilizando el campo amount_company dando como resultado (673,13€ + 141,35€) 814,48€ que es correcto.

Un saludo

pedrobaeza committed:
[FIX] l10n_es_aeat_sii: not_in_amount_total should be currency compatible

Tax amount is in the currency of the invoice, but total amount is on
company currency, so we need to convert the amount if discountable.

Fixes #1552

pedrobaeza commented:
Deberías revisar la rama 12.0...

73444252ec3eb4f08b7683c2e1f31934db60d885 lo soluciona en esa rama.

pedrobaeza commented:
Veo que tu parche echa mano de un campo que no está en v12, pero convendría asegurar que lo incluye el estándar, y que contiene el valor que debe contener.

RodrigoBM commented:
Buenas @pedrobaeza,

Si te refieres al campo amount_company  este campo lo crea el propio modulo [l10n_es_aeat_sii](https://github.com/OCA/l10n-spain/blob/41ca62da4bd0618bca9e02bd6d85d1d717f48ab5/l10n_es_aeat_sii/models/account_invoice_tax.py#L15) tanto en la V11 como V12.

En las pruebas que hice los datos que cogía si que eran correctos. Si que es verdad que tu solución es mejor ya que al usar amount_total sigues teniendo en cuenta amount_rounding cosa que con el campo amount_company solo es la transformación del campo amount a la moneda de la compañia.

Pues cierro PR y hago un cherry pick de tu commit.

Un saludo

pedrobaeza commented:
Sí, ya recuerdo también por qué lo utilicé así: en v13 no hay más account.invoice.tax, y puse esa solución más "universal".

RodrigoBM commented:
Pues lo dicho @pedrobaeza voy hacer unas pruebas con tu commit en la V11 y si va bien creo cherry pick y cierro este PR, muchas gracias

RodrigoBM commented:
Buenas @pedrobaeza,

El problema que veo para la V11 es que no esta disponible el método _convert en "res.currency" por lo cual te propongo un cambio parecido pero que seguiría funcionando en V12 que seria lo siguiente.

```
currency = self.currency_id.with_context(
    date=self._get_currency_rate_date(),
     company_id=self.company_id.id)
amount = currency.compute(
    tax_line.amount_total,
     self.company_id.currency_id,
    )
```
Que te parece el cambio, un saludo

RodrigoBM committed:
[FIX] 11.0 l10n_es_aeat_sii: amount total tax line incorrect
Add cherry pick 73444252ec3eb4f08b7683c2e1f31934db60d885 and chage funcionality to work in V11

This FIX solves the information sent to the SII in invoices that have a different currency. If you have an invoice with:
Base imponible: 605,16£
Cuota de IVA: 127,08£
Our company is Spanish so we have to report these amounts in Euros:
Base imponible: 0,899030 / 605,16£ = 673,13€
Cuota de IVA: 0,899030 / 127,08£ = 141,35€
Without this FIX we would be sending to the SII a total amount of (673,13€ + 127,08£) 800,21€. This is not correct, because the amount of 673,13 is reported in euros but we are adding the VAT in GBP 127,08.
With this correction the amount that we are left to add as VAT fee is the currency of the company using the amount_company field resulting in (673,13€ + 141,35€) 814,48€ which is correct.

RodrigoBM commented:
Buenas @pedrobaeza he añadido tu commit y he cambiado la funcionalidad como te comento, no se si esto es lo mas correcto para incluir el cambio.

Un saludo

pedrobaeza commented:
/ocabot merge patch

RodrigoBM commented:
Gracias @pedrobaeza
